contributing/CODING_STANDARDS.md

```diff
@@ -92,51 +46,67 @@ Sometimes the test is trivial, in that the unit is small and only has
 one code path. In that case shoehorning a test case description into a
 `when`/`then` style may be less readable than a simpler ad-hoc
 alternative. Best judgement should be used, bearing in mind readability
-and consistency tradeoffs.
+and consistency trade-offs.
 
-### IDE configuration
+## C++
 
-To aid in conforming to our coding style a [`.editorconfig`](https://editorconfig.org/)
-file is included in the root of the repository, which has wide support
-across various IDEs. For supported IDEs this config file can override
-the default settings for inspections and formatting refactors.
+### Classes
 
-In particular, more granular settings are included for IntelliJ-based
-IDEs (for example, PyCharm) via the `ij_*` extensions.
+C++ classes that represent system components with reference semantics
+(as opposed to 'value' types) should inherit
+`std::enable_shared_from_this` and define a peer `Ptr` alias using
```

Comment:
Do you think we need the alias? If we also alias `enable_shared_from_this` is is likely we'd ever be able to seamlessly change what we were doing? That mechanisms seems very specific to the STL pointers.

Reply:
Since `enable_shared_from_this` is not templated on the type of smart pointer, if we swap out `SharedPtr` for a different implementation, then `enable_shared_from_this` will definitely break (unless, perhaps, if the type we swapped to can be constructed from a `std::shared_ptr`, but that seems unlikely).

So I think we should either alias both or alias neither. If we alias `enable_shared_from_this` we could conceivably swap it out with a drop-in replacement.  It's very unlikely that we will, just seems like a good practice.

I'm OK with just removing the aliases completely if that's what you prefer.

Reply:
Capturing offline discussion:

 - We will keep the alias for `openassetio:makeShared` as it is Public API, and means that consumers of OpenAssetIO make use of the ptr aliases and `makeShared` - hiding the internal details. It's odd to use the ptr alias with `std::make_shared`, and we dont really want all host/manager code deciding on which pointer types to use.
 - We'll remove (#460) the internal ptr alias, as as pointed out, not also aliasing `enable_shared_from_this` breaks encapsulation, and its unlikely we'd have a drop-in replacement.
 - In the future (#461) work out how we want public object construction to work.